Add "collector troubleshooting" doc (#2382)
open-telemetry/opentelemetry.io: content/en/docs/collector/troubleshooting.md

---

> title: Troubleshooting

Reviewer: Consider adding a short `description` field explaining what this page is about.

> weight: 32

Reviewer: ```suggestion
weight: 25
```

The current weight leaves this page in a strange place:

> <img width="305" alt="image" src="https://user-images.githubusercontent.com/4140793/221566482-da6ed15b-cb6b-46c4-ad0c-ef3cef297a64.png">

---

The OpenTelemetry Collector supports a variety of metrics, logs, and extensions

> for troubleshooting collector health and performance.

Reviewer: The opening paragraph should let the reader know what they'll find in this page.

Detailed recommendations, including common problems, are detailed in the
OpenTelemetry Collector GitHub repo's

> [troubleshooting document](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/troubleshooting.md).

Reviewer 2: I am wondering if this document should live on opentelemetry.io as well? cc @open-telemetry/collector-maintainers ?

Reviewer: Yes! As is already mentioned in the following PR under "Collector":

- #833 

There is valuable information under [opentelemetry-collector/docs](https://github.com/open-telemetry/opentelemetry-collector/tree/main/docs), which (IMHO) would benefit from the increased visibility of being hosted directly on the OTel website.

> [troubleshooting document](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/troubleshooting.md).

Reviewer: I'd suggest moving the content of this paragraph (lines 9 to 11) to the end, under a section entitled (something like) **More**.

---
title: Troubleshooting
description: Recommendations for troubleshooting the collector
weight: 25
---

> This document describes some options when troubleshooting the health or
> performance of The OpenTelemetry Collector. The Collector provides a variety of
> metrics, logs, and extensions for debugging issues.

Reviewer: ```suggestion
This page describes some options when troubleshooting the health or
performance of the OpenTelemetry Collector. The Collector provides a variety of
metrics, logs, and extensions for debugging issues.
```

## Sending test data

For certain types of issues, particularly verifing configuration and debugging
network issues, it can be helpful to send a small amount of data to a collector

> configured to output to local logs. This is detailed in
> ["Local exporters" in the collector configuration documentation](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/troubleshooting.md#local-exporters).

Reviewer: ```suggestion
configured to output to local logs. For details, see
[Local exporters](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/troubleshooting.md#local-exporters).
```

## Checklist for debugging complex pipelines

It can be difficult to isolate problems when telemetry flows through multiple
collectors and networks. For each "hop" of telemetry data through a collector or
other component in your telemetry pipeline, it’s important to verify the
following:

- Are there error messages in the logs of the collector?
- How is the telemetry being ingested into this component?
- How is the telemetry being modified (i.e. sampling, redacting) by this
  component?
- How is the telemetry being exported from this component?
- What format is the telemetry in?
- How is the next hop configured?
- Are there any network policies that prevent data from getting in or out?

### More

> Detailed recommendations, including common problems, are in the OpenTelemetry
> Collector GitHub repo's
> [troubleshooting document](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/troubleshooting.md).

Reviewer: ```suggestion
For detailed recommendations, including common problems, see 
[Troubleshooting](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/troubleshooting.md) from the Collector repo.
```
